http-clients/netty-nio-client/src/main/java/software/amazon/awssdk/http/nio/netty/internal/LastHttpContentHandler.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright 2010-2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.http.nio.netty.internal;
+
+import static software.amazon.awssdk.http.nio.netty.internal.ChannelAttributeKey.LAST_HTTP_CONTENT_RECEIVED_KEY;
+
+import io.netty.channel.ChannelHandler;
+import io.netty.channel.ChannelHandlerContext;
+import io.netty.channel.ChannelInboundHandlerAdapter;
+import io.netty.handler.codec.http.LastHttpContent;
+import software.amazon.awssdk.annotations.SdkInternalApi;
+import software.amazon.awssdk.utils.Logger;
+
+/**
+ * Marks {@code ChannelAttributeKey.LAST_HTTP_CONTENT_RECEIVED_KEY} if {@link LastHttpContent} is received.
+ */
+@SdkInternalApi
+@ChannelHandler.Sharable
+public final class LastHttpContentHandler extends ChannelInboundHandlerAdapter {
+    private static final LastHttpContentHandler INSTANCE = new LastHttpContentHandler();
+    private static final Logger logger = Logger.loggerFor(LastHttpContent.class);
+
+    @Override
+    public void channelRead(ChannelHandlerContext ctx, Object msg) {
+        if (msg instanceof LastHttpContent) {
+            logger.debug(() -> "Received LastHttpContent " + ctx.channel());
+            ctx.channel().attr(LAST_HTTP_CONTENT_RECEIVED_KEY).set(true);
+        }
+
+        ctx.fireChannelRead(msg);
+    }
+
+    public static LastHttpContentHandler create() {
```

Comment:
nit: `getInstance()`

Reply:
Oh I thought our naming convention for static factory method is `create`.

https://github.com/aws/aws-sdk-java-v2/blob/master/docs/design/FavorStaticFactoryMethods.md#naming-conventions

Reply:
Believe that only applies when a new instance is returned for each call.

Reply:
I think we use that for new instance as well as singleton instance in other places.

https://github.com/aws/aws-sdk-java-v2/blob/4c0e42ed66975de42d68df9d508ffd9584d4428f/core/auth/src/main/java/software/amazon/awssdk/auth/credentials/DefaultCredentialsProvider.java#L58-L59

Reply:
Seems misleading to me but it's fine if there's existing classes doing this I guess.